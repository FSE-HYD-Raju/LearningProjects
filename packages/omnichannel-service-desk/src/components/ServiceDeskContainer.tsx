import * as React from "react";
import {
	CharacteristicsComparisonModalContainer,
	CustomerListDropdownContainer,
	OcModal,
	OcTabLink,
	User,
	ContextType,
	FormattedMessage,
	contextTypesValidationMap,
	ComparisonFooterContainer,
	ProductOfferingConfigurationModalContainer,
	NavBarSearchMenu,
	Customer,
	InstallationTimeConfig,
	CustomerCase, OcButton, OcButtonType, ProductPath,
	CustomerBasketsContainer,
} from "omnichannel-common-pos";
import Helmet from "react-helmet";
import StartCustomerCase from "./StartCustomerCase";
import { get, isEmpty } from "lodash";
import * as classnames from "classnames";
import messages from "./ServiceDesk.messages";
import { RouteComponentProps } from "react-router";
import SalesOrganizationModalContainer from "./salesRep/SalesOrganizationModalContainer";
import InstallationTimeConfigurationModal from "./modals/InstallationTimeConfigurationModal";

const ServiceDeskContainerRoutes = require("../routes/ServiceDeskContainerRoutes");

interface ServiceDeskContainerStateProps {
	toolmode: boolean;
	toolmodeIndividualId: string | null;
	user?: User;
	activeCustomerCase: CustomerCase;
	showNoCustomerWarning: boolean;
	showBlacklistedCustomerWarning: boolean;
	singleTermCustomers: Array<any>; // should be Array<Person>
	searchingForSingleCustomer?: boolean;
	showCustomerCreationModal?: boolean;
	showInstallationTimeConfigurationModal: boolean;
	installationTimeConfig?: InstallationTimeConfig;
	posRoutes?: React.ComponentType<any>;
	searchConfigs: string;
	consulValuesLoaded: boolean;
	enableNavbarInToolmode: boolean;
}

interface ServiceDeskContainerActionProps {
	actions: {
		getCustomerById: (customerId: string) => void;
		createNewCustomerCase: (saledRepId?: string) => void;
		showCustomerCreationModal: (keepCustomer: boolean) => void;
		searchCustomerWithSingleTerm: (term: string, criteria: string) => void;
		setCustomer: (customer: Customer, fetchUserBaskets: boolean) => void;
		cancelAddProduct: () => void;
		getAgreements: (customerId: string) => void;
		toggleInstallationTimeConfigurationModal: (show: boolean, config?: InstallationTimeConfig) => void;
		setInputtedCharacteristic: (path: ProductPath, key: string, value: string) => void;
		resetConfigurableInstallationTime: () => void;
		hideNoCustomerWarning: () => void;
	};
}

interface ServiceDeskContainerState {
	customerSearchInput?: string;
	showRegistrationModal?: boolean;
	customerSearchVisible?: boolean;
	criteria: string;
}

type ServiceDeskContainerProps = ServiceDeskContainerStateProps & ServiceDeskContainerActionProps & RouteComponentProps<any>;

class ServiceDeskContainer extends React.Component<ServiceDeskContainerProps, ServiceDeskContainerState> {
	static contextTypes = contextTypesValidationMap;

	constructor(props: ServiceDeskContainerProps, context: ContextType) {
		super(props, context);
		this.state = {
			customerSearchInput: "",
			criteria: "name"
		};
	}

	componentWillMount() {
		const { toolmode, toolmodeIndividualId } = this.props;
		if (toolmode) {
			this.initToolMode(toolmodeIndividualId);
		}
	}

	async initToolMode(individualId: string | null) {
		const salesRepId = this.props.user && this.props.user.id;
		if (individualId) {
			await this.props.actions.getCustomerById(individualId);
		}
		this.props.actions.createNewCustomerCase(salesRepId);
		if (individualId) {
			this.props.actions.getAgreements(individualId);
		}

		if (this.props.location.pathname) {
			this.props.history.push(this.props.location.pathname);
		}
	}

	handleCriteriaChange = (val: string) => {
		this.setState({
			...this.state,
			criteria: val
		});
	};

	handleCustomerSearchModalInput = (e: any) => {
		this.setState({
			customerSearchInput: e.target.value
		});
	};

	handleCustomerSearch = (e: any) => {
		e.preventDefault();
		if (this.state.customerSearchInput) {
			this.props.actions.searchCustomerWithSingleTerm(this.state.customerSearchInput, this.state.criteria);

			this.setState({
				customerSearchVisible: true
			});
		}
	};

	closeCustomerSearch = () => {
		this.setState({
			customerSearchVisible: false
		});
	};

	selectCustomer = (customer: any) => { // should be Person or User here
		if (!isEmpty(customer)) {
			this.props.actions.setCustomer(customer, true);
			this.setState(
				{
					customerSearchInput: "",
					customerSearchVisible: false
				},
			);
			this.props.actions.hideNoCustomerWarning();
		}
	};

	renderSalesOrganizationModal = () => {
		return <SalesOrganizationModalContainer flux={this.context.flux} />;
	};

	renderInstallationTimeConfigurationModal = () => {
		const {
			toggleInstallationTimeConfigurationModal,
			setInputtedCharacteristic,
			resetConfigurableInstallationTime,
		} = this.props.actions;

		return (
			<InstallationTimeConfigurationModal
				showInstallationTimeConfigurationModal={this.props.showInstallationTimeConfigurationModal}
				toggleInstallationTimeConfigurationModal={toggleInstallationTimeConfigurationModal}
				installationTimeConfig={this.props.installationTimeConfig}
				setInputtedCharacteristic={setInputtedCharacteristic}
				resetConfigurableInstallationTime={resetConfigurableInstallationTime}
			/>
		);
	};

	renderUnidentifiedCustomerModal = () => {
		// NOTE: Fix for visual bug when this component is shown with compare modal.
		const backdropStyle = { zIndex: 1050 };
		return (
			<OcModal
				id="UnidentifiedCustomerModal"
				showModal={this.props.showNoCustomerWarning}
				smallModal={true}
				backdropStyle={backdropStyle}
				title={this.context.intl.formatMessage(messages.cantAddProduct)}
				onClose={() => this.props.actions.cancelAddProduct()}
			>
				<div>
					<h4>
						<FormattedMessage {...messages.serviceDeskUnidentified}/>
					</h4>

					<div className="search-or-create-customer">
						<form onSubmit={this.handleCustomerSearch}>
							<div className="input-group">
								<NavBarSearchMenu
									criteria={this.state.criteria}
									unparsedSearchConfigs={this.props.searchConfigs}
									handleCriteriaChange={this.handleCriteriaChange}
								/>
								<input
									type="search"
									className="form-control"
									id="customerSearchFieldToAddProductToBasket"
									value={this.state.customerSearchInput}
									placeholder={this.context.intl.formatMessage(
										messages.searchCustomerPlaceholder
									)}
									onChange={this.handleCustomerSearchModalInput}
								/>
								<div className="input-group-append">
									<OcButton
										outline={true}
										buttonType={OcButtonType.PRIMARY}
										onClick={this.handleCustomerSearch}
										icon="fa-search"
									/>
								</div>
							</div>
						</form>
						{this.state.customerSearchVisible && (
							<CustomerListDropdownContainer
								singleTermCustomers={this.props.singleTermCustomers}
								selectCustomer={this.selectCustomer}
								closeCustomerSearch={this.closeCustomerSearch}
								showCustomerCreationModal={() => this.props.actions.showCustomerCreationModal(true)}
								searchTerm={this.state.customerSearchInput}
								searchingForSingleCustomer={this.props.searchingForSingleCustomer}
							/>
						)}
						<b className="or">
							<FormattedMessage {...messages.or}/>
						</b>
						<OcButton
							buttonType={OcButtonType.PRIMARY}
							onClick={() => this.props.actions.showCustomerCreationModal(true)}
						>
							<FormattedMessage {...messages.createNewCustomer} />
						</OcButton>
					</div>
				</div>
			</OcModal>
		);
	};

	renderBlacklistedCustomerModal = () => {
		// NOTE: Fix for visual bug when this component is shown with compare modal.
		const backdropStyle = { zIndex: 1050 };
		return (
			<OcModal
				id="blacklistedCustomerModal"
				showModal={this.props.showBlacklistedCustomerWarning}
				smallModal={true}
				backdropStyle={backdropStyle}
				title={this.context.intl.formatMessage(messages.cantAddProduct)}
				onClose={() => this.props.actions.cancelAddProduct()}
			>
				<div>
					<h4>
						<FormattedMessage {...messages.customerBlacklisted}/>
					</h4>
				</div>
			</OcModal>
		);
	};

	renderCustomerBasketsModal = () => {
		return <CustomerBasketsContainer flux={this.context.flux} />;
	};

	renderNavbar = (showToolbarTab: boolean) => (
		<nav className="Pos-Desk-tabs w-nav-primary with-nav-context">
			<ul>
				<li
					className={classnames({
						active: this.props.location && this.props.location.pathname.toLowerCase() === "/servicedesk/shop",
					})}
				>
					<OcTabLink id="servicedesk-shop-link" to="/servicedesk/shop">
						<i className="fa fa-list btn-icon-left" />
						<FormattedMessage {...messages.shop} />
					</OcTabLink>
				</li>
				<li
					className={classnames({
						active: this.props.location && this.props.location.pathname.toLowerCase() === "/servicedesk/customer",
					})}
				>
					<OcTabLink id="servicedesk-customer-link" to="/servicedesk/customer">
						<i className="fa fa-user btn-icon-left" />
						<FormattedMessage {...messages.customer} />
					</OcTabLink>
				</li>
				{
					showToolbarTab && (
						<li
							className={classnames({
								active: this.props.location && this.props.location.pathname.toLowerCase() === "/servicedesk/toolmode",
							})}
						>
							<OcTabLink id="servicedesk-crstb-tab" to="/servicedesk/toolmode">
								<i className="fa fa-wrench btn-icon-left" />
								<FormattedMessage {...messages.toolMode} />
							</OcTabLink>
						</li>
					)}
			</ul>
		</nav>
	);

	render() {
		return (
			<div className="ServiceDesk-container">
				<FormattedMessage {...messages.serviceDesk}>
					{(message: any): any => <Helmet title={message} />}
				</FormattedMessage>

				<FormattedMessage {...messages.singlePointOfContact}>
					{(message: any): any => (
						<Helmet
							meta={[
								{
									name: "description",
									content: `${message}`
								}
							]}
						/>
					)}
				</FormattedMessage>

				{!this.props.activeCustomerCase && (
					<main id="w-app-body">
						<StartCustomerCase
							salesRepId={get(this.props, "user.id")}
							toolmode={this.props.toolmode}
							actions={{
								createNewCustomerCase: this.props.actions.createNewCustomerCase
							}}
						/>
					</main>
				)}

				{this.props.activeCustomerCase &&
					!this.props.toolmode &&
					this.renderNavbar(true)}

				{this.props.activeCustomerCase && this.props.toolmode && this.props.enableNavbarInToolmode && this.renderNavbar(false)}

				{this.props.activeCustomerCase && (
					<ServiceDeskContainerRoutes
						flux={this.context.flux}
						posCheckoutRoute={this.props.posRoutes}
					/>
				)}

				{this.renderUnidentifiedCustomerModal()}
				{this.renderSalesOrganizationModal()}
				{this.renderInstallationTimeConfigurationModal()}
				{this.renderBlacklistedCustomerModal()}
				{this.renderCustomerBasketsModal()}

				<ComparisonFooterContainer/>
				<CharacteristicsComparisonModalContainer match={this.props.match} flux={this.context.flux} />
				<ProductOfferingConfigurationModalContainer flux={this.context.flux} />
			</div>
		);
	}
}

export default ServiceDeskContainer;
export {
	ServiceDeskContainerStateProps,
	ServiceDeskContainerActionProps,
	ServiceDeskContainerProps,
	ServiceDeskContainerState,
};
