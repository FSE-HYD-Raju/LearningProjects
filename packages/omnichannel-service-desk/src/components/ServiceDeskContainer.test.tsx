import * as React from "react";
import {
	attachWithContext,
	mountWithContext,
	shallowWithContext,
	CustomerCase,
	TestUtils,
	mockRouterProps,
	User,
	InstallationTimeConfig,
} from "omnichannel-common-pos";
import ServiceDeskContainer, { ServiceDeskContainerProps } from "./ServiceDeskContainer";
import { ReactWrapper } from "enzyme";

describe("ServiceDeskContainer", () => {

	const redux = TestUtils.mockReduxStore({
		comparison: {},
		category: {
			mainCategories: [],
		},
		productOfferingConfiguration: {
			configurations: {}
		},
		navBar: {
			toolMode: false,
		},
	});

	const context = {
		store: redux,
		flux: {
			actions: {
				CustomerCaseActions: {
					getAgreements: jest.fn(),
				},
				BasketActions: {
					addProductToBasket: jest.fn(),
				},
				UserActions: {
					actAsUser: jest.fn(),
					updateUserDemoGraphicInfo: jest.fn(),
					updateUserAddresses: jest.fn(),
					updateUserPreferences: jest.fn(),
					updateUserPrivacySettings: jest.fn(),
					refreshUser: jest.fn(),
				},
				SalesRepSessionActions: {
					setSelectedOrganization: jest.fn(),
					revertSalesOrganizationAndInventory: jest.fn(),
					getInventories: jest.fn(),
					getOrganizations: jest.fn(),
				}
			},
			stores: {
				ConsulStore: {},
				CustomerStore: {},
				SalesStore: {
					fetchingProducts: false,
				},
				UserStore: {
					user: {
						salesRepUser: {},
					},
					updatingUser: false,
				},
				CustomerCaseStore: {
					activeCustomerCase: {
						attributes: {
							activeCustomer: {
							}
						}
					},
					state: {
						activeCustomerCase: true
					},
					agreements: [],
					getCustomerOffers: [],
					getSidebarNotifications: []
				},
			}
		}
	};

	const minimumProps: ServiceDeskContainerProps = {
		actions: {
			createNewCustomerCase: jest.fn(),
			getCustomerById: jest.fn(),
			showCustomerCreationModal: jest.fn(),
			searchCustomerWithSingleTerm: jest.fn(),
			setCustomer: jest.fn(),
			cancelAddProduct: jest.fn(),
			getAgreements: jest.fn(),
			toggleInstallationTimeConfigurationModal: jest.fn(),
			setInputtedCharacteristic: jest.fn(),
			resetConfigurableInstallationTime: jest.fn(),
			hideNoCustomerWarning: jest.fn(),
			hideBlacklistedCustomerWarning: jest.fn(),
		},
		...mockRouterProps,
		toolmode: false,
		toolmodeIndividualId: "",
		activeCustomerCase: {
			attributes: {
				activeCustomer: {
					id: "asdf"
				}
			}
		} as any as CustomerCase,
		showNoCustomerWarning: false,
		showBlacklistedCustomerWarning: false,
		singleTermCustomers: [],
		showCustomerCreationModal: false,
		showInstallationTimeConfigurationModal: false,
		installationTimeConfig: {} as any as InstallationTimeConfig,
		consulValuesLoaded: false,
		searchConfigs: "",
		enableNavbarInToolmode: false,
	};

	it("succeeds at shallow with minimum props", () => {
		const wrapper = shallowWithContext(<ServiceDeskContainer {...minimumProps} />, { context });
		expect(wrapper).toMatchSnapshot();
	});

	it("succeeds at deep mount with minimum props ", () => {
		const wrapper = mountWithContext(<ServiceDeskContainer {...minimumProps} />, { context });
		wrapper.unmount();
	});

	it("renders tool mode tab if activeCustomerCase.attributes.activeCustomer", () => {
		const props = {
			...minimumProps,
			activeCustomerCase: {
				attributes: {
					activeCustomer: {
						id: "foo"
					}
				}
			} as any as CustomerCase,
		};

		const wrapper = attachWithContext(<ServiceDeskContainer {...props} />, { context });
		expect(wrapper.text()).toContain("Tool mode");
		wrapper.detach();
	});

	it("doesn't render tabs when it is opened in tool mode", () => {
		const props = {
			...minimumProps,
			activeCustomerCase: {
				attributes: {
					activeCustomer: {}
				}
			} as any as CustomerCase,
			toolmode: true
		};

		const wrapper = shallowWithContext(<ServiceDeskContainer {...props} />, { context });
		expect(wrapper.find(".Pos-Desk-tabs")).toHaveLength(0);

		wrapper.unmount();
	});

	it("calls initToolMode() when opened with toolmode and individualId", () => {
		const getCustomerById = minimumProps.actions.getCustomerById;
		const props = {
			...minimumProps,
			user: {
				id: "dummy-id"
			} as any as User,
			toolmode: true,
			toolmodeIndividualId: "sanna"
		};

		mountWithContext(<ServiceDeskContainer {...props} />, { context });
		// Enough to check that both toolmode and toolmodeIndividualId are used properly
		expect(getCustomerById).toHaveBeenCalledWith("sanna");
	});

	it("shows Customer search modal and passes search term and criteria to search function", done => {
		const { getModalContents } = TestUtils;

		const name = "Timo Isokoski";

		const props: ServiceDeskContainerProps = {
			...minimumProps,
			actions: {
				...minimumProps.actions,
				searchCustomerWithSingleTerm: (term, criteria) => {
					expect(term).toEqual(name);
					expect(criteria).toEqual("name");
					done();
				},
			},
			showNoCustomerWarning: true,
		};

		const wrapper = attachWithContext(<ServiceDeskContainer {...props} />, { context });

		const unidentifiedCustomerModal = wrapper.find("OcModal")
			.filterWhere((n: ReactWrapper) => (n.props() as any).id === "UnidentifiedCustomerModal");

		expect(unidentifiedCustomerModal.length).toEqual(1);

		const modalContents = getModalContents(unidentifiedCustomerModal);

		const form = modalContents.find("form");
		const customerSearchField = form.find("#customerSearchFieldToAddProductToBasket").hostNodes();

		customerSearchField.simulate("change", { target: { value: name } });
		form.simulate("submit");

		wrapper.detach();
	});
});
