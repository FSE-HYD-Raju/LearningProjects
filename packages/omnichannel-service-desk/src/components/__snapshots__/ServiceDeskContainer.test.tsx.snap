// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ServiceDeskContainer succeeds at shallow with minimum props 1`] = `
<div
  className="ServiceDesk-container"
>
  <FormattedMessageWrapper
    defaultMessage="Service Desk"
    description="Service desk title"
    id="service-desk-title"
    wrapperType="wrapper"
  >
    <Component />
  </FormattedMessageWrapper>
  <FormattedMessageWrapper
    defaultMessage="Single point of contact for handling your shop and customers"
    description="Service desk description"
    id="service-desk-description"
    wrapperType="wrapper"
  >
    <Component />
  </FormattedMessageWrapper>
  <nav
    className="Pos-Desk-tabs w-nav-primary with-nav-context"
  >
    <ul>
      <li
        className=""
      >
        <OcTabLink
          activeClassName=""
          className=""
          id="servicedesk-shop-link"
          onlyActiveOnIndex={false}
          to="/servicedesk/shop"
        >
          <i
            className="fa fa-list btn-icon-left"
          />
          <FormattedMessageWrapper
            defaultMessage="Shop"
            description="Label for shop link"
            id="service-desk-front-page-shop-link"
            wrapperType="wrapper"
          />
        </OcTabLink>
      </li>
      <li
        className=""
      >
        <OcTabLink
          activeClassName=""
          className=""
          id="servicedesk-customer-link"
          onlyActiveOnIndex={false}
          to="/servicedesk/customer"
        >
          <i
            className="fa fa-user btn-icon-left"
          />
          <FormattedMessageWrapper
            defaultMessage="Customer"
            description="Label for customer link"
            id="service-desk-front-page-customer-link"
            wrapperType="wrapper"
          />
        </OcTabLink>
      </li>
      <li
        className=""
      >
        <OcTabLink
          activeClassName=""
          className=""
          id="servicedesk-crstb-tab"
          onlyActiveOnIndex={false}
          to="/servicedesk/toolmode"
        >
          <i
            className="fa fa-wrench btn-icon-left"
          />
          <FormattedMessageWrapper
            defaultMessage="Tool mode"
            description="Label for toolmode link"
            id="service-desk-tool-mode-link"
            wrapperType="wrapper"
          />
        </OcTabLink>
      </li>
    </ul>
  </nav>
  <ServiceDeskContainerRoutes
    flux={
      Object {
        "actions": Object {
          "BasketActions": Object {
            "addProductToBasket": [MockFunction],
          },
          "CustomerCaseActions": Object {
            "getAgreements": [MockFunction],
          },
          "SalesRepSessionActions": Object {
            "getInventories": [MockFunction],
            "getOrganizations": [MockFunction],
            "revertSalesOrganizationAndInventory": [MockFunction],
            "setSelectedOrganization": [MockFunction],
          },
          "UserActions": Object {
            "actAsUser": [MockFunction],
            "refreshUser": [MockFunction],
            "updateUserAddresses": [MockFunction],
            "updateUserDemoGraphicInfo": [MockFunction],
            "updateUserPreferences": [MockFunction],
            "updateUserPrivacySettings": [MockFunction],
          },
        },
        "reduxStore": Object {
          "dispatch": [Function],
          "getState": [Function],
          "replaceReducer": [Function],
          "subscribe": [Function],
          Symbol(observable): [Function],
        },
        "stores": Object {
          "ConsulStore": Object {},
          "CustomerCaseStore": Object {
            "activeCustomerCase": Object {
              "attributes": Object {
                "activeCustomer": Object {},
              },
            },
            "agreements": Array [],
            "getCustomerOffers": Array [],
            "getSidebarNotifications": Array [],
            "state": Object {
              "activeCustomerCase": true,
            },
          },
          "CustomerStore": Object {},
          "SalesStore": Object {
            "fetchingProducts": false,
          },
          "UserStore": Object {
            "updatingUser": false,
            "user": Object {
              "salesRepUser": Object {},
            },
          },
        },
      }
    }
  />
  <OcModal
    backdropStyle={
      Object {
        "zIndex": 1050,
      }
    }
    hideCloseButton={false}
    id="UnidentifiedCustomerModal"
    onClose={[Function]}
    showModal={false}
    smallModal={true}
    title="Can't add product to basket"
  >
    <div>
      <h4>
        <FormattedMessageWrapper
          defaultMessage="To add products to basket you have to"
          description="Service desk unidentified customer modal, info message"
          id="service-desk-unidentified-customer-modal-information"
          wrapperType="wrapper"
        />
      </h4>
      <div
        className="search-or-create-customer"
      >
        <form
          onSubmit={[Function]}
        >
          <div
            className="input-group"
          >
            <OnClickOutside(NavBarSearchMenu)
              criteria="name"
              eventTypes={
                Array [
                  "mousedown",
                  "touchstart",
                ]
              }
              excludeScrollbar={false}
              handleCriteriaChange={[Function]}
              outsideClickIgnoreClass="ignore-react-onclickoutside"
              preventDefault={false}
              stopPropagation={false}
              unparsedSearchConfigs=""
            />
            <input
              className="form-control"
              id="customerSearchFieldToAddProductToBasket"
              onChange={[Function]}
              placeholder="Search customer"
              type="search"
              value=""
            />
            <div
              className="input-group-append"
            >
              <OcButton
                buttonType="PRIMARY"
                icon="fa-search"
                onClick={[Function]}
                outline={true}
              />
            </div>
          </div>
        </form>
        <b
          className="or"
        >
          <FormattedMessageWrapper
            defaultMessage="or"
            description="Or"
            id="service-desk-unidentified-customer-modal-or"
            wrapperType="wrapper"
          />
        </b>
        <OcButton
          buttonType="PRIMARY"
          onClick={[Function]}
        >
          <FormattedMessageWrapper
            defaultMessage="Create new customer"
            description="service-desk, new customer, create new customer link"
            id="sd-newcustomer-create-new-customer"
            wrapperType="wrapper"
          />
        </OcButton>
      </div>
    </div>
  </OcModal>
  <Connect(SalesOrganizationModal)
    flux={
      Object {
        "actions": Object {
          "BasketActions": Object {
            "addProductToBasket": [MockFunction],
          },
          "CustomerCaseActions": Object {
            "getAgreements": [MockFunction],
          },
          "SalesRepSessionActions": Object {
            "getInventories": [MockFunction],
            "getOrganizations": [MockFunction],
            "revertSalesOrganizationAndInventory": [MockFunction],
            "setSelectedOrganization": [MockFunction],
          },
          "UserActions": Object {
            "actAsUser": [MockFunction],
            "refreshUser": [MockFunction],
            "updateUserAddresses": [MockFunction],
            "updateUserDemoGraphicInfo": [MockFunction],
            "updateUserPreferences": [MockFunction],
            "updateUserPrivacySettings": [MockFunction],
          },
        },
        "reduxStore": Object {
          "dispatch": [Function],
          "getState": [Function],
          "replaceReducer": [Function],
          "subscribe": [Function],
          Symbol(observable): [Function],
        },
        "stores": Object {
          "ConsulStore": Object {},
          "CustomerCaseStore": Object {
            "activeCustomerCase": Object {
              "attributes": Object {
                "activeCustomer": Object {},
              },
            },
            "agreements": Array [],
            "getCustomerOffers": Array [],
            "getSidebarNotifications": Array [],
            "state": Object {
              "activeCustomerCase": true,
            },
          },
          "CustomerStore": Object {},
          "SalesStore": Object {
            "fetchingProducts": false,
          },
          "UserStore": Object {
            "updatingUser": false,
            "user": Object {
              "salesRepUser": Object {},
            },
          },
        },
      }
    }
  />
  <InstallationTimeConfigurationModal
    installationTimeConfig={Object {}}
    resetConfigurableInstallationTime={[MockFunction]}
    setInputtedCharacteristic={[MockFunction]}
    showInstallationTimeConfigurationModal={false}
    toggleInstallationTimeConfigurationModal={[MockFunction]}
  />
  <OcModal
    backdropStyle={
      Object {
        "zIndex": 1050,
      }
    }
    hideCloseButton={false}
    id="blacklistedCustomerModal"
    onClose={[Function]}
    showModal={false}
    smallModal={true}
    title="Can't add product to basket"
  >
    <div>
      <h4>
        <FormattedMessageWrapper
          defaultMessage="Customer blacklisted, not possible to proceed"
          description="service-desk, title for dialog which appears when customer is blacklisted"
          id="service-desk-blacklisted-customer"
          wrapperType="wrapper"
        />
      </h4>
    </div>
  </OcModal>
  <CustomerBasketContainer
    flux={
      Object {
        "actions": Object {
          "BasketActions": Object {
            "addProductToBasket": [MockFunction],
          },
          "CustomerCaseActions": Object {
            "getAgreements": [MockFunction],
          },
          "SalesRepSessionActions": Object {
            "getInventories": [MockFunction],
            "getOrganizations": [MockFunction],
            "revertSalesOrganizationAndInventory": [MockFunction],
            "setSelectedOrganization": [MockFunction],
          },
          "UserActions": Object {
            "actAsUser": [MockFunction],
            "refreshUser": [MockFunction],
            "updateUserAddresses": [MockFunction],
            "updateUserDemoGraphicInfo": [MockFunction],
            "updateUserPreferences": [MockFunction],
            "updateUserPrivacySettings": [MockFunction],
          },
        },
        "reduxStore": Object {
          "dispatch": [Function],
          "getState": [Function],
          "replaceReducer": [Function],
          "subscribe": [Function],
          Symbol(observable): [Function],
        },
        "stores": Object {
          "ConsulStore": Object {},
          "CustomerCaseStore": Object {
            "activeCustomerCase": Object {
              "attributes": Object {
                "activeCustomer": Object {},
              },
            },
            "agreements": Array [],
            "getCustomerOffers": Array [],
            "getSidebarNotifications": Array [],
            "state": Object {
              "activeCustomerCase": true,
            },
          },
          "CustomerStore": Object {},
          "SalesStore": Object {
            "fetchingProducts": false,
          },
          "UserStore": Object {
            "updatingUser": false,
            "user": Object {
              "salesRepUser": Object {},
            },
          },
        },
      }
    }
  />
  <Connect(ComparisonFooter) />
  <Connect(CharacteristicsComparisonModal)
    flux={
      Object {
        "actions": Object {
          "BasketActions": Object {
            "addProductToBasket": [MockFunction],
          },
          "CustomerCaseActions": Object {
            "getAgreements": [MockFunction],
          },
          "SalesRepSessionActions": Object {
            "getInventories": [MockFunction],
            "getOrganizations": [MockFunction],
            "revertSalesOrganizationAndInventory": [MockFunction],
            "setSelectedOrganization": [MockFunction],
          },
          "UserActions": Object {
            "actAsUser": [MockFunction],
            "refreshUser": [MockFunction],
            "updateUserAddresses": [MockFunction],
            "updateUserDemoGraphicInfo": [MockFunction],
            "updateUserPreferences": [MockFunction],
            "updateUserPrivacySettings": [MockFunction],
          },
        },
        "reduxStore": Object {
          "dispatch": [Function],
          "getState": [Function],
          "replaceReducer": [Function],
          "subscribe": [Function],
          Symbol(observable): [Function],
        },
        "stores": Object {
          "ConsulStore": Object {},
          "CustomerCaseStore": Object {
            "activeCustomerCase": Object {
              "attributes": Object {
                "activeCustomer": Object {},
              },
            },
            "agreements": Array [],
            "getCustomerOffers": Array [],
            "getSidebarNotifications": Array [],
            "state": Object {
              "activeCustomerCase": true,
            },
          },
          "CustomerStore": Object {},
          "SalesStore": Object {
            "fetchingProducts": false,
          },
          "UserStore": Object {
            "updatingUser": false,
            "user": Object {
              "salesRepUser": Object {},
            },
          },
        },
      }
    }
    match={
      Object {
        "isExact": true,
        "params": Object {},
        "path": "",
        "url": "",
      }
    }
  />
  <Connect(withMsisdnConfiguration.)
    flux={
      Object {
        "actions": Object {
          "BasketActions": Object {
            "addProductToBasket": [MockFunction],
          },
          "CustomerCaseActions": Object {
            "getAgreements": [MockFunction],
          },
          "SalesRepSessionActions": Object {
            "getInventories": [MockFunction],
            "getOrganizations": [MockFunction],
            "revertSalesOrganizationAndInventory": [MockFunction],
            "setSelectedOrganization": [MockFunction],
          },
          "UserActions": Object {
            "actAsUser": [MockFunction],
            "refreshUser": [MockFunction],
            "updateUserAddresses": [MockFunction],
            "updateUserDemoGraphicInfo": [MockFunction],
            "updateUserPreferences": [MockFunction],
            "updateUserPrivacySettings": [MockFunction],
          },
        },
        "reduxStore": Object {
          "dispatch": [Function],
          "getState": [Function],
          "replaceReducer": [Function],
          "subscribe": [Function],
          Symbol(observable): [Function],
        },
        "stores": Object {
          "ConsulStore": Object {},
          "CustomerCaseStore": Object {
            "activeCustomerCase": Object {
              "attributes": Object {
                "activeCustomer": Object {},
              },
            },
            "agreements": Array [],
            "getCustomerOffers": Array [],
            "getSidebarNotifications": Array [],
            "state": Object {
              "activeCustomerCase": true,
            },
          },
          "CustomerStore": Object {},
          "SalesStore": Object {
            "fetchingProducts": false,
          },
          "UserStore": Object {
            "updatingUser": false,
            "user": Object {
              "salesRepUser": Object {},
            },
          },
        },
      }
    }
  />
</div>
`;
